src/__tests__/try-path.test.ts

```diff
@@ -95,6 +95,58 @@ describe("mapping-entry", () => {
     ]);
   });
 
+  it("should include paths with ending .js removed that matches requested module", () => {
+    const result = getPathsToTry(
+      [".ts", ".tsx"],
+      [
+        {
+          pattern: "longest/pre/fix/*",
+          paths: [join("/absolute", "base", "url", "foo2", "*")],
+        },
+      ],
+      "longest/pre/fix/bar.js"
+    );
+
+    expect(result).toEqual(
+      expect.arrayContaining([
+        {
+          type: "extension",
+          path: join("/absolute", "base", "url", "foo2", "bar.ts"),
+        },
+        {
+          type: "extension",
+          path: join("/absolute", "base", "url", "foo2", "bar.tsx"),
+        },
+      ])
+    );
+  });
+
+  it("should include paths with ending .mjs removed that matches requested module", () => {
+    const result = getPathsToTry(
+      [".ts", ".tsx", ".mts", ".mtsx"],
```

Comment:
So I'm not sure `.mtsx` is a file extension that exists - not to the best of my knowledge anyway. What _is_ a valid spec extension is `.cts` and `.cjs` (for CommonJS files). See https://nodejs.org/api/esm.html#interoperability-with-commonjs

Reply:
And I use the work "spec" in a sense that this is the `stability: 2` decision made by Node.js.

Reply:
We should cover all the bases that Vite does

https://github.com/vitejs/vite/blob/632fedf87fbcb81b2400571886faf8a8b92376e4/packages/vite/src/node/__tests__/utils.spec.ts#L124-L144

Reply:
TIL :)

Reply:
> So I'm not sure .mtsx is a file extension that exists

I was just testing you :P (but in reality I just copy pasted the same thing and added an m before it at night)

> We should cover all the bases that Vite does

I'll try to adjust the code to support the ones listed in the linked spec file (probably tomorrow)